### PR TITLE
ui: refactor get api params in forms

### PR DESCRIPTION
**Pull request by shwstppr**
### Description

Refactor code getting API params for APIs in UI forms.
Added a new util plugin in plugins.js

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [x] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by shwstppr** [comment]
Based on testing and feasibility either of this PR or #5063 can be merged.

@blueorangutan ui

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by shwstppr** [comment]
@blueorangutan ui

**Review by utchoang — APPROVED** [review]
LGTM!

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by shwstppr** [comment]
@blueorangutan ui

**Comment by shwstppr** [comment]
@blueorangutan ui

File: ui/src/views/iam/AddUser.vue [hunk 1]
```diff
@@ -222,7 +222,19 @@ export default {
...
+    this.apiConfig = this.$store.getters.apis.createUser || {}
+    this.apiParams = {}
+    this.apiConfig.params.forEach(param => {
+      this.apiParams[param.name] = param
+    })
+    this.apiConfig = this.$store.getters.apis.authorizeSamlSso || { params: [] }
+    this.apiConfig.params.forEach(param => {
+      this.apiParams[param.name] = param
+    })
+>>>>>>> 4.15
```
Comment (DaanHoogland) [inline ui/src/views/iam/AddUser.vue]: diff in a commit can't be good

Comment (shwstppr) [reply ui/src/views/iam/AddUser.vue]: @DaanHoogland isn't that fixed already in the last commit?

Comment (DaanHoogland) [reply ui/src/views/iam/AddUser.vue]: i see now, it is outdated. Don't know what happened but fine ;)

**Comment by rhtyd** [comment]
@shwstppr is this ready for review/testing/merge?

**Comment by shwstppr** [comment]
@rhtyd yes, this is ready

**Review by rhtyd — APPROVED** [review]

**Pull request merged**